### PR TITLE
[Localization] Fix fallback should not be stored in case of 0 or false value

### LINKED ISSUE #11984: [Bug]: When generating a localized view, the fallback language is used if its a checkbox and its value is false
### Expected behavior

Say we have an object with a localized checkbox field `enabled`. We also have a set of languages with fallbacks, e.g. `de_AT` with a fallback to `de`. 

When I set the `enabled` value to `false` (or `0`) within an objects `de_AT`, and to `true` (or `1`) within `de`, the returned value (from the database view) should be `0`. 
So when I run something like this:
```
        $employeeListing = new ContactPerson\Listing();
        $employeeListing->setLocale('de_AT');
        // only show employees that are enabled for this locale
        $employeeListing->setCondition("enabled = 1");
```
it should NOT return this object.

### Actual behavior

However, this object IS returned.

### Steps to reproduce

And this is the reason.
The generated SQL view would look something like this:
```
select `database`.`object_query_9`.`oo_id`                                                     AS `oo_id`,
       `database`.`object_query_9`.`oo_classId`                                                AS `oo_classId`,
       `database`.`object_query_9`.`oo_className`                                              AS `oo_className`,
       `database`.`objects`.`o_id`                                                             AS `o_id`,
       `database`.`objects`.`o_parentId`                                                       AS `o_parentId`,
       `database`.`objects`.`o_type`                                                           AS `o_type`,
       `database`.`objects`.`o_key`                                                            AS `o_key`,
       `database`.`objects`.`o_path`                                                           AS `o_path`,
       `database`.`objects`.`o_index`                                                          AS `o_index`,
       `database`.`objects`.`o_published`                                                      AS `o_published`,
       `database`.`objects`.`o_creationDate`                                                   AS `o_creationDate`,
       `database`.`objects`.`o_modificationDate`                                               AS `o_modificationDate`,
       `database`.`objects`.`o_userOwner`                                                      AS `o_userOwner`,
       `database`.`objects`.`o_userModification`                                               AS `o_userModification`,
       `database`.`objects`.`o_classId`                                                        AS `o_classId`,
       `database`.`objects`.`o_className`                                                      AS `o_className`,
       `database`.`objects`.`o_childrenSortBy`                                                 AS `o_childrenSortBy`,
       `database`.`objects`.`o_childrenSortOrder`                                              AS `o_childrenSortOrder`,
       `database`.`objects`.`o_versionCount`                                                   AS `o_versionCount`,
       `de_AT`.`ooo_id`                                                                                 AS `ooo_id`,
       if(((`de_AT`.`enabled` is null) or (`de_AT`.`enabled` = '')), `de`.`enabled`, `de_AT`.`enabled`) AS `enabled`
from (
    (
        (
            `database`.`object_query_9` 
            join `database`.`objects`
                on ((`database`.`objects`.`o_id` = `database`.`object_query_9`.`oo_id`))
        ) 
        left join `database`.`object_localized_query_9_de_AT` `de_AT`
            on (((0 <> 1) and (`database`.`object_query_9`.`oo_id` = `de_AT`.`ooo_id`)))
    ) 
    left join `database`.`object_localized_query_9_de` `de`
        on (((0 <> 1) and (`database`.`object_query_9`.`oo_id` = `de`.`ooo_id`)))
)
```
and the problem lies within this statement: 
```
if(((`de_AT`.`enabled` is null) or (`de_AT`.`enabled` = '')), `de`.`enabled`, `de_AT`.`enabled`) AS `enabled`
```
because if the value stored in `object_localized_query_9_de_AT`.`enabled` is false (or `0`) and thus
```
`de_AT`.`enabled` = ''
```
will be read as `true`, so the view will actually use the fallback language setting
```
`de`.`enabled`
```
which returns `1`

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #11984

### CODE REVIEW

File: tests/Model/DataType/LocalizedFieldTest.php [hunk 1]
```diff
@@ -92,4 +93,48 @@ public function testLocalizedFieldInsideFieldCollection()
...
+        $object->setLcheckbox(true, 'en');
+        $object->setLnumber(123, 'en');
+
+        //de values
+        $object->setLinput('TestDE', 'de');
+        $object->setLcheckbox(true, 'de');
+        $object->setLnumber(0, 'de');
+
+        $object->save();
+
```
Comment (fashxp) [inline tests/Model/DataType/LocalizedFieldTest.php:116]: I would add checks here to make sure getter and listings work properly

File: tests/Model/DataType/LocalizedFieldTest.php [hunk 2]
```diff
@@ -92,4 +93,48 @@ public function testLocalizedFieldInsideFieldCollection()
...
+        //empty de values check fallback
+        $object->setLinput('', 'de');
+        $object->setLcheckbox(false, 'de');
```
Comment (fashxp) [inline tests/Model/DataType/LocalizedFieldTest.php]: can't we set `null` here ... as it says we empty the values?

File: tests/Model/DataType/LocalizedFieldTest.php [hunk 3]
```diff
@@ -92,4 +93,48 @@ public function testLocalizedFieldInsideFieldCollection()
...
+        $object->save();
+
+        $object = DataObject\Unittest::getById($object->getId(), true);
+
+        $this->assertEquals('TestEN', $object->getLinput('de'));
+        $this->assertEquals(0, $object->getLnumber('de')); //fallback should not return, num 0 is set
+
+        //fallback for checkbox is not supported anymore
+        //$this->assertEquals(true, $object->getLcheckbox('de'));
```
Comment (fashxp) [inline tests/Model/DataType/LocalizedFieldTest.php]: when value is actively set to `false`, I would not expect that fallback value is taken ... only when value is set to `null`.

File: tests/Model/DataType/LocalizedFieldTest.php (hunk 2 above)
Comment (fashxp) [reply tests/Model/DataType/LocalizedFieldTest.php]: we also need to set number to null and check if fallback works properly

File: tests/Model/DataType/LocalizedFieldTest.php [hunk 4]
```diff
@@ -92,4 +93,48 @@ public function testLocalizedFieldInsideFieldCollection()
...
+
+        //asset listing works with fallback value
+        $listing = new DataObject\Unittest\Listing();
+        $listing->setLocale('de');
+
+        $listing->setCondition("linput = 'TestEN'");
+        $this->assertEquals(1, count($listing->load()), 'Expected 1 item for fallback en condition');
+
+        $listing->setCondition("lcheckbox = '1'");
+        $this->assertEquals(0, count($listing->load()), 'Expected 0 item for fallback en condition');
```
Comment (fashxp) [inline tests/Model/DataType/LocalizedFieldTest.php]: also test for number field filtering

**Review by fashxp — CHANGES_REQUESTED** [review]

File: tests/Model/DataType/LocalizedFieldTest.php (hunk 2 above)
Comment (dvesh3) [reply tests/Model/DataType/LocalizedFieldTest.php]: I used the value possible to set from Admin UI

Comment (dvesh3) [reply tests/Model/DataType/LocalizedFieldTest.php]: checking both cases now, when setting `null` & `false`(special case)

**Review by fashxp — APPROVED** [review]

**Pull request merged**